packages/gatsby/src/utils/api-node-docs.js

```diff
@@ -82,9 +82,38 @@ exports.createPagesStatefully = true
  *
  * See also the documentation for [`createNode`](/docs/actions/#createNode).
  * @example
- * exports.sourceNodes = ({ actions }) => {
+ * const crypto = require(`crypto`)
+ *
+ * exports.sourceNodes = ({ actions, createNodeId }) => {
  *   const { createNode } = actions
- *   // Create nodes here.
+ *
+ *   // Data can come from anywhere, but for now create it manually
+ *   const myData = {
+ *     key: 123,
+ *     foo: `The foo field of my node`,
+ *     bar: `Baz`
+ *   }
+ *
+ *   const nodeContent = JSON.stringify(myData)
+ *   const nodeContentDigest = crypto
```

Comment:
Unrelated, but I think you mentioned it too... we use this algorithm _all_ over the place, so it's probably worth passing that to various APIs or something, e.g.

```js
const createContentDigest = obj => crypto
  .createHash(`md5`)
  .update(obj)
  .digest(`hex`)
```

Reply:
:+1: to that, here's an issue https://github.com/gatsbyjs/gatsby/issues/8587